recipes/libpng/all/conanfile.py

```diff
@@ -1,9 +1,11 @@
-import os
-from conans import CMake, tools
 from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
 from conan.tools.scm import Version
-from conan.tools.files import apply_conandata_patches, get, rmdir, replace_in_file
+from conan.tools.files import apply_conandata_patches, get, rmdir, replace_in_file, copy, rm
+from conan.tools.microsoft import is_msvc
 from conan.tools.build.cross_building import cross_building
+import os
+
 
 required_conan_version = ">=1.50.2"
```

Comment:
this conditiion is partially correct, since it breaks for conan 1.51.0

Reply:
True, need to check if we can avoid directly from required_conan_version, otherwise, will need to use validation step

Reply:
Looking at conan source code, `required_conan_version = " >=1.50.2 <1.51.0 || >=1.51.2"`should work.

Reply:
Done. Added `required_conan_version = ">=1.50.2 <1.51.0 || >=1.51.2"`. Thank you @SpaceIm